MediaBrowser.Providers/Lyric/LrcLyricProvider.cs

```diff
@@ -0,0 +1,196 @@
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using LrcParser.Model;
+using LrcParser.Parser;
+using MediaBrowser.Controller.Entities;
+using MediaBrowser.Controller.Lyrics;
+using MediaBrowser.Controller.Resolvers;
+using Microsoft.Extensions.Logging;
+
+namespace MediaBrowser.Providers.Lyric;
+
+/// <summary>
+/// LRC Lyric Provider.
+/// </summary>
+public class LrcLyricProvider : ILyricProvider
+{
+    private readonly ILogger<LrcLyricProvider> _logger;
+
+    private readonly LyricParser _lrcLyricParser;
+
+    private static readonly IReadOnlyList<string> _acceptedTimeFormats = new string[] { "HH:mm:ss", "H:mm:ss", "mm:ss", "m:ss" };
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="LrcLyricProvider"/> class.
+    /// </summary>
+    /// <param name="logger">Instance of the <see cref="ILogger"/> interface.</param>
+    public LrcLyricProvider(ILogger<LrcLyricProvider> logger)
+    {
+        _logger = logger;
+        _lrcLyricParser = new LrcParser.Parser.Lrc.LrcParser();
+    }
+
+    /// <inheritdoc />
+    public string Name => "LrcLyricProvider";
+
+    /// <summary>
+    /// Gets the priority.
+    /// </summary>
+    /// <value>The priority.</value>
+    public ResolverPriority Priority => ResolverPriority.First;
+
+    /// <inheritdoc />
+    public IReadOnlyCollection<string> SupportedMediaTypes { get; } = new[] { "lrc", "elrc" };
+
+    /// <summary>
+    /// Opens lyric file for the requested item, and processes it for API return.
+    /// </summary>
+    /// <param name="item">The item to to process.</param>
+    /// <returns>If provider can determine lyrics, returns a <see cref="LyricResponse"/> with or without metadata; otherwise, null.</returns>
+    public LyricResponse? GetLyrics(BaseItem item)
+    {
+        string? lyricFilePath = LyricInfo.GetLyricFilePath(this, item.Path);
```

Comment:
If you make it an extension method, this call could be changed to this
```suggestion
        string? lyricFilePath = this.GetLyricFilePath(item.Path);
```